client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/logging/future/BucketFuture.java

```diff
@@ -0,0 +1,61 @@
+package org.kaaproject.kaa.client.logging.future;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.*;
+
+public class BucketFuture<T> implements Future<T> {
```

Comment:
As this feature is for 0.8.0 release, it means what Java 8 may be used.
So I suggest to use https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html.